builtin/providers/google/resource_google_project_services.go

```diff
@@ -0,0 +1,200 @@
+package google
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"google.golang.org/api/servicemanagement/v1"
+)
+
+func resourceGoogleProjectServices() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGoogleProjectServicesCreate,
+		Read:   resourceGoogleProjectServicesRead,
+		Update: resourceGoogleProjectServicesUpdate,
+		Delete: resourceGoogleProjectServicesDelete,
+
+		Schema: map[string]*schema.Schema{
+			"project": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"services": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+		},
+	}
+}
+
+func resourceGoogleProjectServicesCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	pid := d.Get("project").(string)
+	err := updateProjectServices(d, config, pid)
+	if err != nil {
+		return err
+	}
+	d.SetId(pid)
+	return resourceGoogleProjectServicesRead(d, meta)
+}
+
+func resourceGoogleProjectServicesRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	// Put the services defined in the manifest in a map
+	known := make(map[string]bool)
+	if r, ok := d.GetOk("services"); ok {
+		for _, knownS := range r.(*schema.Set).List() {
+			known[knownS.(string)] = true
+		}
+	}
+
+	services, err := config.clientServiceMan.Services.List().ConsumerId("project:" + d.Id()).Do()
+	if err != nil {
+		return err
+	}
+
+	// Inventory and record actual services state, but only care about
+	// services that are in the manifest
```

Comment:
I'm nervous about the drift this could open us up to--services changing outside the manifest means the manifest no longer documents which services are enabled. This also means our diffs get weird and we can no longer accurately remove services from a project. Kinda concerned about the implications of this.